### PR TITLE
D3ASIM-3447: Port common client code to d3a-interface

### PR DESCRIPTION
**Pull request by Hassan754**

### CODE REVIEW

File: d3a_api_client/websocket_device.py [hunk 1]
```diff
@@ -1,17 +1,11 @@
...
+
+from d3a_interface.client_connections.websocket_connection import WebsocketMessageReceiver
 from d3a_interface.utils import wait_until_timeout_blocking
-from d3a_api_client.utils import retrieve_jwt_key_from_server
-from d3a_api_client.constants import WEBSOCKET_ERROR_THRESHOLD_SECONDS, WEBSOCKET_MAX_CONNECTION_RETRIES, \
-    WEBSOCKET_WAIT_BEFORE_RETRY_SECONDS
 
 
-class WebsocketMessageReceiver:
+class AssetWebsocketMessageReceiver(WebsocketMessageReceiver):
```
Comment (spyrostz) [inline d3a_api_client/websocket_device.py:8]: I think it is better to name this DeviceWebsocketMessageReceiver, in order to retain the same naming as the file name.

**Review by spyrostz — APPROVED** [review]
LGTM, however the device-measurement-subscriber repo needs to be updated too though, IIRC the WebsocketMessageReceiver is also used in this repo.

File: setup.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from setuptools import setup, find_packages
 from d3a_api_client import __version__
 
-target_branch = os.environ.get('BRANCH', 'master')
+target_branch = os.environ.get('BRANCH', 'feature/D3ASIM-3447')
```
Comment (hannesdiedrich) [inline setup.py:5]: Reminder for changing this before merging to master

**Review by hannesdiedrich — DISMISSED** [review]
LGTM

**Review by faizan2590 — APPROVED** [review]

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**